scripts/mvftoms.py

```diff
@@ -281,6 +328,77 @@ def _cp_index(a1, a2, pol):
+    dump_ants = split_ants
+
+    if not set(dump_ants) <= set(avail_ants):
+        raise ValueError("One or more antennas cannot be found in the dataset. "
+                         "You requested {0:s}, but only {1:s} are available.".format(
+                         ", ".join(["'{}'".format(f) for f in dump_ants]),
+                         ", ".join(["'{}'".format(f) for f in avail_ants])))
+
+    if len(dump_ants) == 0:
+       print('User antenna criterion resulted in empty database, nothing to be done. '
+             'Perhaps you wanted to select from the following: {}'.format(
+             ", ".join(["'{}'".format(f) for f in avail_ants])))
+
+    print('Per user request the following antennae will be dumped: {}'.format(
+        ", ".join(["'{}'".format(f) for f in dump_ants])))
+
+    # select a subset of targets
+    avail_fields = [f.name for f in dataset.catalogue.targets]
+    dump_fields = options.field if len(options.field) != 0 else \
+                  avail_fields
+
+    # some users may specify comma-separated lists although we said the switch should
+    # be specified multiple times. Put in a guard to split comma separated lists as well
+    split_fields = []
+    for f in dump_fields:
+        split_fields += [x.strip() for x in f.split(",")]
+    dump_fields = split_fields
+
+    if not set(dump_fields) <= set(avail_fields):
+        raise ValueError("One or more fields cannot be found in the dataset. "
+                         "You requested {0:s}, but only {1:s} are available.".format(
+                         ", ".join(["'{}'".format(f) for f in dump_fields]),
+                         ", ".join(["'{}'".format(f) for f in avail_fields])))
+
+    if len(dump_fields) == 0:
+       print('User field criterion resulted in empty database, nothing to be done. '
+             'Perhaps you wanted to select from the following: {}'.format(
+             ", ".join(["'{}'".format(f) for f in avail_fields])))
+
+    print('Per user request the following fields will be dumped: {}'.format(
+        ", ".join(["'{}'".format(f) for f in dump_fields])))
+
+    dataset.select(targets=dump_fields)
+
+    # get a set of user selected available tracking scans, ignore slew scans
+    avail_tracks = map(lambda x: x[0],
+                     filter(lambda x: x[1] == 'track',
+                            list(dataset.scans())))
+
+    dump_scans = options.scans if options.scans else \
+                 avail_tracks
+    dump_scans = list(set(dump_scans).intersection(set(avail_tracks)))
+    if len(dump_scans) == 0:
+        print('User scan criterion resulted in empty database, nothing to be done. '
+              'Perhaps you wanted to select from the following: {}'.format(
+               ", ".join(map(str, avail_tracks))))
+        sys.exit(0)
+    print('Per user request the following scans will be dumped: {}'.format(
+        ", ".join(map(str, dump_scans))))
+
+    dataset.select(scans=dump_scans, targets=dump_fields)
```

Comment:
This select seems superfluous.

Reply:
ah good catch I forgot to remove that one